crates/symbolicator-service/src/services/module_lookup.rs

```diff
@@ -383,22 +389,60 @@ impl ModuleLookup {
             .collect()
     }
 
-    /// This looks up the source of the given line, plus `n` lines above/below.
-    pub fn get_context_lines(
+    /// Update the frame with source context, if available. In case the source code references an
+    /// URL that's not yet loaded, returns `SetContextLinesResult::SourceMissing(url)`.
+    pub(crate) fn set_source_context(
         &self,
         debug_sessions: &HashMap<usize, Option<ObjectDebugSession<'_>>>,
-        frame: &RawFrame,
-    ) -> Option<(Vec<String>, String, Vec<String>)> {
+        frame: &mut RawFrame,
+    ) -> Option<SetSourceContextResult> {
```

Comment:
IMO it might be simpler to return an `Option<Url>` here.
You only ever match the `Some(SourceMissing(...))` part.